source/last-array-element.d.ts

```diff
@@ -0,0 +1,22 @@
+/**
+Extracts the type of the last element of an array.
+
+Use case: Defining the return type of functions that extract the last element of an array, for example [lodash.last](https://lodash.com/docs/4.17.15#last).
+
+@example
+```
+import {LastArrayElement} from 'type-fest';
+
+declare function lastOf<V extends any[], L extends LastArrayElement<V>>(array: V): L;
+const array = ['foo', 2];
+typeof lastOf(array); // -> number;
+```
+*/
+export type LastArrayElement<V> =
+	V extends []
+		? never
+		: V extends [string]
```

Comment:
Do you need this check? Why `[string]` specifically?

Could the whole implementation reduce to something like

```typescript
export type LastArrayElement<V extends unknown[]> =
  V extends [infer _, ...infer Tail] ? LastArrayElement<Tail> : never
```

Reply:
Good catch and smart reimplementation. Applied.